### PR TITLE
Add redux actions for deleting a site

### PR DESCRIPTION
**Pull request by oskosk**
#### Changes introduced by this PR.

* Adds a `deleteSite` action in `state/sites/actions`.
* Adds an `isDeletingSite` selector.
* Adds notices for deleting a site in the notices middleware.
  * It also considers a special case of notice when the user is trying to delete a site that still has upgrades associated at the moment of deletion attempt.
* Updates the `receiveDeletedSite` action creator signature to accept a `siteId` instead of a `site` object. 

**Note**: This structure updates an already existing `receiveDeletedSite()` action creator. The signature for `receiveDeletedSite()` is updated here to accept just a `siteId` as argument. Given that `receiveDeletedSite()` was also used by the `confirm-cancel-domain` and `remove-purchase` components, this PR updates them to conform to the new signature.

#### Testing instructions

1. Run the unit tests on `state/sites`. (`npm run test-client client/state/sites` ).
1. Run the unit tests on `state/selectors`. (`npm run test-client client/state/selectors` ).

#### Why

Refactoring the `DeleteSite` component (#13468) in order to remove usage of sites-list from the `/settings/delete-site` page, demanded a new deleteSite action to be created, so I ended up splitting the PR.

### CODE REVIEW

**Comment by oskosk** [comment]
Thanks @tyxla . I've addressed the observations.

* Moved `isDeletingSite` selector to `state/selectors` .
* Updated jsdoc description for selector arguments.
* Move the action type that was messing with alphabetical order.
* Use `keyedReducer()`.
* Show the error returned from the API in notices instead of fixed text.
* Add same id to notices
* return dispatch from notices mid.dleware handlers related to site delete.

File: client/state/sites/actions.js [hunk 1]
```diff
@@ -14,19 +22,18 @@ import {
...
 	SITES_UPDATE
 } from 'state/action-types';
-import { omit } from 'lodash';
 
 /**
  * Returns an action object to be used in signalling that a site has been
  * deleted.
  *
- * @param  {Object} site Site received
+ * @param  {Number} siteId Site id received
```
Comment (oskosk) [inline client/state/sites/actions.js]: I addressed the description here, but about the alignment issue, looks like you were comparing two mutually exclusive lines. One's out for the other and the return description was already aligned to the parameter.

Comment (tyxla) [inline client/state/sites/actions.js]: Actually, as far as I can see right now, they're still not aligned. Compare lines 30 and 31 in client/state/sites/actions.js and you'll see what I mean:

```
 * @param  {Number} siteId ID of deleted site
 * @return {Object}      Action object
```

Comment (oskosk) [inline client/state/sites/actions.js]: haha I messed it up. fixed now

File: client/state/sites/test/actions.js [hunk 2]
```diff
@@ -191,4 +196,56 @@ describe( 'actions', () => {
...
 		} );
 	} );
+
+	describe( 'deleteSite()', () => {
+		useNock( nock => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
+				.post( '/rest/v1.1/sites/2916284/delete' )
+				.reply( 200, {
+					ID: 2916284
```
Comment (oskosk) [inline client/state/sites/test/actions.js]: By specific response, yo mean the body ? The delete endpoint returns the deleted site object. this was minimally mocking the actual reply.

Comment (tyxla) [inline client/state/sites/test/actions.js]: Yeah, but if we're not using it, it's enough to just return a `200` code with no actual response.

Comment (tyxla) [inline client/state/sites/test/actions.js]: Feel free to ignore 👍

**Pull request merged**